Update dotty to 0.24.0-RC1

Following https://github.com/zio/izumi-reflect/issues/33, it's time to update to dotty 0.24.0-RC1 & izumi-reflect 1.0.0-M3

@adamgfraser 
I reproduced that:
```
Error:(12, 10) Exception occurred while executing macro expansion.
java.lang.AssertionError: assertion failed
	at dotty.DottyPredef$.assertFail(DottyPredef.scala:16)
	at dotty.tools.dotc.ast.tpd$.polyDefDef(tpd.scala:239)
	at dotty.tools.dotc.ast.tpd$.DefDef(tpd.scala:225)
	at dotty.tools.dotc.ast.tpd$.DefDef(tpd.scala:222)
	at dotty.tools.dotc.tastyreflect.FromSymbol$.classDef(FromSymbol.scala:33)
	at dotty.tools.dotc.tastyreflect.FromSymbol$.definitionFromSym(FromSymbol.scala:16)
	at dotty.tools.dotc.tastyreflect.ReflectionCompilerInterface.Symbol_tree(ReflectionCompilerInterface.scala:1672)
	at dotty.tools.dotc.tastyreflect.ReflectionCompilerInterface.Symbol_tree(ReflectionCompilerInterface.scala:1671)
	at scala.tasty.Reflection$SymbolOps$.tree(Reflection.scala:2161)
	at izumi.reflect.dottyreflection.Inspector.inspectTType(DbInspector.scala:55)
	at izumi.reflect.dottyreflection.Inspector.inspectTree(DbInspector.scala:121)
	at izumi.reflect.dottyreflection.Inspector.buildTypeRef(DbInspector.scala:42)
	at izumi.reflect.dottyreflection.TypeInspections$.apply(DbInspector.scala:265)
	at izumi.reflect.dottyreflection.Inspect$.inspectAny(DbInspector.scala:21)
  inspect[(Double, Double)]
```
Something happened to Tuple's `ClassDef#tree`. It's either a regression in dotty - could you check if current nightly wokrs? or a more fundamental change to how we should inspect types (e.g. switch from trees to other apis, there were some relevant functions exposed recently), i've added ticket for this at https://github.com/zio/izumi-reflect/issues/38

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -10,7 +10,7 @@ scala_213: &scala_213
   SCALA_VERSION: 2.13.1
 
 scala_dotty: &scala_dotty
-  SCALA_VERSION: 0.23.0-RC1
+  SCALA_VERSION: 0.26.0-RC1
 
 jdk_8: &jdk_8
   JDK_VERSION: 8

--- a/project/BuildHelper.scala
+++ b/project/BuildHelper.scala
@@ -53,7 +53,7 @@ object BuildHelper {
       buildInfoObject := "BuildInfo"
     )
 
-  val dottyVersion = "0.23.0-RC1"
+  val dottyVersion = "0.26.0-RC1"
 
   val dottySettings = Seq(
     // Keep this consistent with the version in .circleci/config.yml
@@ -83,7 +83,7 @@ object BuildHelper {
   )
 
   val scalaReflectSettings = Seq(
-    libraryDependencies ++= Seq("dev.zio" %%% "izumi-reflect" % "1.0.0-M2")
+    libraryDependencies ++= Seq("dev.zio" %%% "izumi-reflect" % "1.0.0-M5")
   )
 
   // Keep this consistent with the version in .core-tests/shared/src/test/scala/REPLSpec.scala

--- a/sbt
+++ b/sbt
@@ -37,7 +37,7 @@ set -o pipefail
 declare -r sbt_release_version="1.3.13"
 declare -r sbt_unreleased_version="1.4.0-M1"
 
-declare -r latest_dotty="0.23.0-RC1"
+declare -r latest_dotty="0.26.0-RC1"
 declare -r latest_213="2.13.3"
 declare -r latest_212="2.12.12"
 declare -r latest_211="2.11.12"

